crates/esplora/tests/async_ext.rs

```diff
@@ -1,5 +1,6 @@
 use bdk_chain::spk_client::{FullScanRequest, SyncRequest};
 use bdk_chain::{ConfirmationBlockTime, TxGraph};
+#[cfg(feature = "async")]
```

Comment:
Would it accomplish the same thing if we enable these features as dev-dependencies?

```rs
[dev-dependencies]
bdk_esplora = { path = ".", features = ["async", "blocking"] }
```

Reply:
I was wondering about it, if either (1) using `#[cfg(feature = "..."]`, (2) your suggestion, or using [`required-features`](https://doc.rust-lang.org/cargo/reference/cargo-targets.html#the-required-features-field) would be the best approach.

Reply:
I guess a better question is should these tests run when doing `cargo test --workspace --no-default-features` or is it fine to skip them as long as they are picked up when doing `--all-features`

Reply:
@ValuedMammal Using the `required-features` approach would look like this in `Cargo.toml`, for example: 

``` toml
[[test]]
name = "blocking"
path = "tests/blocking_ext.rs"
required-features = ["blocking"]

[[test]]
name = "async"
path = "tests/async_ext.rs"
required-features = ["async"]
```

Reply:
Yes I think that sounds reasonable

Reply:
Sweet! I updated both commits, and it's ready for re-review.